crates/core_arch/src/x86/avx512f.rs

```diff
@@ -9270,6 +10519,77 @@ pub unsafe fn _mm512_kxor(a: __mmask16, b: __mmask16) -> __mmask16 {
     transmute(kxorw(a, b))
 }
 
+/// Compute the bitwise NOT of 16-bit mask a, and store the result in k.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=knot_mask16&expand=3233)
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(all(test, not(target_os = "macos")), assert_instr(not))] // generate normal not code instead of knotw
```

Comment:
Why are you special-casing macos here?

Reply:
Can you post the fully assembly output you are getting on macOS? I find it extremely strange that it is behaving differently from Linux.